docs/refresh-token.md

```diff
@@ -0,0 +1,159 @@
+
+
+## Refreshing Access Token
+
+The following is an example of middleware that handles the token refresh process based on the sample provided by [@Good-man](https://github.com/Good-man) on this issue https://github.com/okta/okta-aspnet/issues/130.
+
+### Startup.cs
+
+```csharp
+        public void Configuration(IAppBuilder app)
+        {
+            app.SetDefaultSignInAsAuthenticationType(CookieAuthenticationDefaults.AuthenticationType);
+
+            app.UseCookieAuthentication(new CookieAuthenticationOptions());
+            
+            // Adding the Refresh token middleware
+            var oktaDomain = ConfigurationManager.AppSettings["okta:OktaDomain"];
+            var authServerId = ConfigurationManager.AppSettings["okta:AuthorizationServerId"];
+            var tokenEndpoint = $"{oktaDomain}/oauth2/{authServerId}/v1/token";
+
+            app.Use(typeof(RefreshTokenMiddleware), new RefreshTokenMiddlewareOptions
+            {
+                TokenEndpoint = tokenEndpoint,
+                ClientId = ConfigurationManager.AppSettings["okta:ClientId"],
+                ClientSecret = ConfigurationManager.AppSettings["okta:ClientSecret"],
+            });
+            
+            // Adding Okta Mvc
+            app.UseOktaMvc(new OktaMvcOptions()
+            {
+                OktaDomain = oktaDomain,
+                ClientId = ConfigurationManager.AppSettings["okta:ClientId"],
+                ClientSecret = ConfigurationManager.AppSettings["okta:ClientSecret"],
+                AuthorizationServerId = authServerId,
+                RedirectUri = ConfigurationManager.AppSettings["okta:RedirectUri"],
+                PostLogoutRedirectUri = ConfigurationManager.AppSettings["okta:PostLogoutRedirectUri"],
+                GetClaimsFromUserInfoEndpoint = true,
+
+                Scope = new List<string> { "openid", "profile", "email", "offline_access" },
+            });
+        }
+```
+
+
+### RefreshTokenMiddleware.cs
+
```

Comment:
I suggest adding a brief comment on the purpose of the required classes and logic.